diesel/src/sqlite/connection/statement_iterator.rs

```diff
@@ -161,12 +161,12 @@ impl<'stmt, 'query> Iterator for StatementIterator<'stmt, 'query> {
                     )
                 }
             }
-            NotStarted(_) => unreachable!(
-                "You've reached an impossible internal state. \
-                             If you ever see this error message please open \
-                             an issue at https://github.com/diesel-rs/diesel \
-                             providing example code how to trigger this error."
-            ),
+            NotStarted(_s) => {
+                // we likely got an error while executing the other
+                // `NotStarted` branch above. In this case we just want to stop
+                // iterating here
+                None
```

Comment:
This seems reasonable to me. Either we've already returned some error or we've already returned `None`. In either case, returning `None` forever after is good.